### DISCUSSION TITLE
Implement an unique ID / Number to bookings

### DISCUSSION THREAD

**Discussion by carlosvirreira** [Shelf-nu/shelf.nu#620] [Ideas]
Similar to how Github handles issues. Adding an ```#ID``` to each booking allows for natural language to be used when referring to a booking. 

```Can you check out 503 for me?```

Also, it allows for users who have ```reserved``` items that are blocking a ```checkout``` from happening to be able to see the booking ```#ID``` to 'debug/fix' the issue.

**Comment by DonKoko** [comment]
I agree. I think its a great addition. Do we want some specific naming structure or just a number. I saw in some designs it was like Booking-1

**Reply by carlosvirreira** [reply]
I think the Github approach is pretty efficient. A hashtag and a number. Technically not sure if it is possible without heavily rebuilding our breadcrumbs or url structure.

**Reply by DonKoko** [reply]
Okey so a few things about the github approach:
1. They use it for multiple model types. So their numeration includes issues as well as PRs. I am not sure this is a good idea for us.
2. So you want to also change the url structure, so instead of id we use this thing? Its possible, will just be quite inconsistent with the rest of the app.

**Reply by carlosvirreira** [reply]
Well caught 

1. I did not notice that the issues AND discussions (and for that regard, many other areas) are sharing the same numeration logic. 
2. No I do not - I just thought this could be one reason why adding the ID only would not work. On GH they use /repo_id/area/ID | In shelf we do not have the /user_id/ or /workspace_id/ on the URL
